sdk/core/azure-core/inc/azure/core/cryptography/hash.hpp

```diff
@@ -45,13 +45,14 @@ namespace Azure { namespace Core { namespace Cryptography {
      */
     virtual std::vector<uint8_t> OnFinal(const uint8_t* data, std::size_t length) = 0;
 
-  public:
+  protected:
     /**
-     * @brief Construct a default instance of #Azure::Core::Cryptography::Hash.
+     * @brief Constructs a default instance of `%Hash`.
```

Comment:
Just to confirm, isn't this a breaking change, it needs to be called out in the CL, and please try to avoid such subtle changes in large doc updates.

Why do we need to make this change? Please revert changing from public to protected in this PR and open a separate PR for this, and let's discuss the reasons/merits there.

Reply:
Since the instances of the class can't be created because it is abstract, only the classes deriving from it will have access to this constructor, for which `protected` is sufficient.

Reply:
I think we should leave it, it is not technically a breaking change, since you can't call a public constructor of an abstract class, other than deriving from it, for which `protected` constructor is accessible.

Reply:
I am planning to not take this to the follow up PR.

Reply:
Do we need to define a ctor at all in this case? For example, BodyStream is an abstract class, that doesn't have a ctor.

Reply:
You are right - we don't need to (we would've need if if we needed a custom one, but since we are using the default - we don't need to define it at all

Reply:
Or, if there were custom constructors with parameters, but we also wanted to have a parameterless one.

Reply:
I added that to #2390.